src/DependencyInjection/Compiler/FilterResolverRegistryCompilerPass.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace SonataMediaBundle\DependencyInjection\Compiler;
+
+use Liip\ImagineBundle\DependencyInjection\Compiler\AbstractCompilerPass;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+class FilterResolverRegistryCompilerPass extends AbstractCompilerPass
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $tags = $container->findTaggedServiceIds('liip_imagine.cache.resolver');
+
+        if (\count($tags) > 0 && $container->hasDefinition('sonata.media.liip_imagine.filter_resolver_registry')) {
+            $manager = $container->getDefinition('sonata.media.liip_imagine.filter_resolver_registry');
+
+            foreach ($tags as $id => $tag) {
+                $manager->addMethodCall('addResolver', [$tag[0]['resolver'], new Reference($id)]);
+                $this->log($container, 'Registered cache resolver: %s', $id);
```

Comment:
Link for the lazy: https://github.com/liip/LiipImagineBundle/blob/master/DependencyInjection/Compiler/AbstractCompilerPass.php#L24